src/main/java/io/anserini/rerank/lib/RocchioReranker.java

```diff
@@ -0,0 +1,211 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.rerank.lib;
+
+import io.anserini.analysis.AnalyzerUtils;
+import io.anserini.rerank.Reranker;
+import io.anserini.rerank.RerankerContext;
+import io.anserini.rerank.ScoredDocuments;
+import io.anserini.util.FeatureVector;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static io.anserini.search.SearchCollection.BREAK_SCORE_TIES_BY_DOCID;
+import static io.anserini.search.SearchCollection.BREAK_SCORE_TIES_BY_TWEETID;
+
+public class RocchioReranker implements Reranker {
+  private static final Logger LOG = LogManager.getLogger(RocchioReranker.class);
+
+  private final Analyzer analyzer;
+  private final String field;
+
+  private final int fbTerms;
+  private final int fbDocs;
+  private final float originalQueryWeight;
+  private final boolean outputQuery;
+  private final boolean filterTerms;
+
+  public RocchioReranker(Analyzer analyzer, String field, int fbTerms, int fbDocs, float originalQueryWeight, boolean outputQuery, boolean filterTerms) {
+    this.analyzer = analyzer;
+    this.field = field;
+    this.fbTerms = fbTerms;
+    this.fbDocs = fbDocs;
+    this.originalQueryWeight = originalQueryWeight;
+    this.outputQuery = outputQuery;
+    this.filterTerms = filterTerms;
+  }
+
+  @Override
+  public ScoredDocuments rerank(ScoredDocuments docs, RerankerContext context) {
+    assert(docs.documents.length == docs.scores.length);
+
+    IndexSearcher searcher = context.getIndexSearcher();
+    IndexReader reader = searcher.getIndexReader();
+
+    FeatureVector qfv = FeatureVector.fromTerms(AnalyzerUtils.analyze(analyzer, context.getQueryText())).scaleToUnitL1Norm();
```

Comment:
Interesting. `scaleToUnitL1Norm()` isn't part of the original definition of Rocchio. I wonder what happens if you remove?

Reply:
scaleToUnitL1Norm rises the score a bit, so I still kept it there.

Reply:
did you remove the `scaleToUnitL1Norm()` on the document vector also? You do so below....

Reply:
No I didn't remove the both scaleToUnitL1Norm() for query and document vector